core/src/main/java/org/infinispan/interceptors/locking/AbstractTxLockingInterceptor.java

```diff
@@ -148,20 +169,60 @@ private InvocationStage checkPendingAndLockAllKeys(TxInvocationContext<?> ctx, V
       PendingLockPromise pendingLockPromise =
          pendingLockManager.checkPendingTransactionsForKeys(ctx, keys, lockTimeout, TimeUnit.MILLISECONDS);
       if (pendingLockPromise.isReady()) {
-         return lockAllAndRecord(ctx, command, keys, lockTimeout);
+         return lockAllAndRecord(ctx, command, keys, lockTimeout).andExceptionallyMakeStage(ctx, command, lockLeakCheck);
       }
 
       return pendingLockPromise.toInvocationStage().thenApplyMakeStage(ctx, command, ((rCtx, rCommand, rv) -> {
          long remaining = pendingLockPromise.getRemainingTimeout();
-         return lockAllAndRecord(ctx, command, keys, remaining);
+         return lockAllAndRecord(rCtx, rCommand, keys, remaining).andExceptionallyMakeStage(rCtx, rCommand, lockLeakCheck);
       }));
    }
 
-   void releaseLockOnTxCompletion(TxInvocationContext ctx) {
+   void releaseLockOnTxCompletion(TxInvocationContext<?> ctx) {
       boolean shouldReleaseLocks = ctx.isOriginLocal() &&
             !partitionHandlingManager.isTransactionPartiallyCommitted(ctx.getGlobalTransaction());
       if (shouldReleaseLocks) {
          lockManager.unlockAll(ctx);
       }
    }
+
+   private Object checkForLockLeak(InvocationContext ctx, VisitableCommand cmd, Throwable throwable) throws Throwable {
+      LockTimeoutException.findLockTimeoutException(throwable)
+            .ifPresent(checkGtxLockLeak);
+      throw throwable;
+   }
+
+   private void checkGlobalTransactionForLockLeak(LockTimeoutException timeoutException) {
+      if (!(timeoutException.getLockOwner() instanceof GlobalTransaction)) {
+         return;
+      }
+      final GlobalTransaction gtx = (GlobalTransaction) timeoutException.getLockOwner();
+      final Log log = getLog();
+      if (isRemote(gtx)) {
+         if (log.isTraceEnabled()) {
+            log.tracef("GlobalTransaction %s is remote to this node.", gtx);
+         }
+         return;
+      }
+      LocalTransaction tx = transactionTable.getLocalTransaction(gtx);
+      if (tx != null) {
+         final String msg = timeoutException.getMessage();
+         final long duration = timeService.timeDuration(tx.getCreationTime(), TimeUnit.MILLISECONDS);
+         final Collection<Object> keys = tx.getAffectedKeys();
+         log.warnLockOwnerInfoOnTimeout(msg, Util.prettyPrintTime(duration), keys);
```

Comment:
Is this log adding any new information to the logs we already have in InvocationContextInterceptor/TransactionCoordinator/BasePerCacheInboundInvocationHandler?

Reply:
yes: how long the transaction is active and the locked keys.

Reply:
Yeah, I should have noticed that :)

I was just thinking that we already log the lock timeout acquisition errors in multiple places, so I would have preferred if we enhanced the log messages we already have instead of adding one more WARN message that will get lost among all the ERROR messages that we log for the same LockTimeoutException (and usually when there's one LTE, there are many LTE around the same time).